specification/ews/SecretSyncController.Management/azurekeyvaultsecretproviderclass.tsp

```diff
@@ -0,0 +1,77 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@azure-tools/typespec-azure-core";
+import "./common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.Core;
+using Azure.ResourceManager;
+
+@armProviderNamespace("Microsoft.SecretSyncController")
+namespace Microsoft.SecretSyncController;
+
+@doc("The AzureKeyVaultSecretProviderClass resource.")
+model AzureKeyVaultSecretProviderClass
+  is TrackedResource<AzureKeyVaultSecretProviderClassProperties> {
+  ...ResourceNameParameter<
+    AzureKeyVaultSecretProviderClass,
+    NamePattern = "^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+  >;
+  ...ExtendedLocationProperty;
+}
+
+@@maxLength(AzureKeyVaultSecretProviderClass.name, 253);
+
+@doc("The properties of the AzureKeyVaultSecretProviderClass.")
+model AzureKeyVaultSecretProviderClassProperties {
+  // Azure Key Vault Provider properties
+  // https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/usage/
+  @doc("The name of the Azure Key Vault to sync secrets from.")
+  keyvaultName: string;
+
+  @doc("The user assigned managed identity client ID that should be used to access the Azure Key Vault.")
+  @pattern("^[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}$")
+  clientId: string;
+
+  @doc("The Azure Active Directory tenant ID that should be used for authenticating requests to the Azure Key Vault.")
+  @pattern("^[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}$")
+  tenantId: string;
+
+  @doc("Objects defines the desired state of synced K8s secret objects")
+  objects?: string;
+
+  @doc("Provisioning state of the AzureKeyVaultSecretProviderClass instance.")
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+}
+
+@armResourceOperations
+interface AzureKeyVaultSecretProviderClasses {
+  @doc("Gets the properties of an AzureKeyVaultSecretProviderClass instance.")
+  get is ArmResourceRead<AzureKeyVaultSecretProviderClass>;
+
+  @doc("Creates, or updates, an AzureKeyVaultSecretProviderClass instance.")
+  createOrUpdate is ArmResourceCreateOrUpdateAsync<AzureKeyVaultSecretProviderClass>;
+
+  @doc("Updates an AzureKeyVaultSecretProviderClass instance.")
+  update is ArmCustomPatchAsync<
+    AzureKeyVaultSecretProviderClass,
+    Azure.ResourceManager.Foundations.ResourceUpdateModel<
+      AzureKeyVaultSecretProviderClass,
+      AzureKeyVaultSecretProviderClassProperties
+    >
+  >;
+
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes" "RPaaS can return both async and sync responses."
+  #suppress "deprecated" "RPaaS can return both async and sync responses."
+  @doc("Deletes an AzureKeyVaultSecretProviderClass instance.")
+  delete is ArmResourceDeleteAsync<AzureKeyVaultSecretProviderClass>;
```

Comment:
`ArmResourceDeleteAsync` was deprecate on 7/11/23:

https://github.com/Azure/typespec-azure/blob/main/docs/release-notes/release-2023-07-11.md#azure-toolstypespec-azure-resource-manager-deprecations

Are you sure you should still be using it for your spec?  Or should you instead use `ArmResourceDeleteWithoutOkAsync`, as recommended in the deprecation note?

Reply:
@rkmanda, @markcowl: In the Teams thread, you seemed to think this was a bug in RPaaS that should be fixed, and specs should keep using `ArmResourceDeleteWithoutOkAsync`.

https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/delete-resource.md

Reply:
We have since spoken with John and agreed this is expected behaviour

Reply:
As getting the public preview available is high priority for us, I can remove the `ArmResourceDeleteAsync` change if that is going to slow this approval down.

However, without it the spec doesn't match the behaviour we are seeing from RPaaS and K8sBridge, and resulted in errors when we were testing the generated SDKs. I think that it something you should discuss with the RPaaS team, as this is not behaviour we control and we have been told it is intentional.